### PR TITLE
Add same behavior for filters and sorting as in other types

### LINKED ISSUE #1265: Overriding of filters does not work as expected
```csharp


        [Fact]
        public void Bind_Filter_ImplicitlyTwice()
        {
            // arrange
            // act
            ISchema schema = CreateSchema(
                new FilterInputType<Foo>(descriptor =>
                {
                    descriptor
                        .BindFieldsExplicitly()
                        .Filter(x => x.Bar)
                        .BindFiltersExplicitly()
                        .BindFiltersImplicitly();
                    descriptor
                        .BindFieldsExplicitly()
                        .Filter(x => x.Bar)
                        .BindFiltersExplicitly()
                        .AllowEquals();
                }));

            // assert
            schema.ToString().MatchSnapshot();
        }

```

yields 
```graphql
input FooFilter {
  AND: [FooFilter!]
  bar: Boolean
  bar_not: Boolean
  OR: [FooFilter!]
}
```

### PR DESCRIPTION
**Pull request by PascalSenn**
Resolves #1265

### CODE REVIEW

File: src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            }
+            if (valueFactory == null)
+            {
+                throw new ArgumentNullException(nameof(valueFactory));
+            }
+
+            FilterFieldDescriptorBase fieldDescriptor =
+                fields.FirstOrDefault(t => t.Definition.Property.Equals(propertyInfo));
+
+            if (fieldDescriptor is { })
```
Comment (michaelstaib) [inline src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs]: Shouldn't this lead to a schema exception? How would it be possible to have a different descriptor for the same property on the same filter type?

File: src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            {
+                throw new ArgumentNullException(nameof(valueFactory));
+            }
+
+            FilterOperationDescriptorBase operationDescriptor =
+                fields.FirstOrDefault(t => t.Definition.Operation.Kind.Equals(operationKind));
+
+            if (operationDescriptor is { })
+            {
+                if (operationDescriptor is T descritorOfT)
```
Comment (michaelstaib) [inline src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs]: same here

Comment (PascalSenn) [reply src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs]: I agree on this one, i need to check the code further

File: src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs (hunk 1 above)
Comment (PascalSenn) [reply src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs]: The problem solved  here is this one. 

https://github.com/ChilliCream/hotchocolate/blob/master/src/Core/Types.Filters/IgnoredFilterFieldDescriptor.cs

```
descirptor =>  descriptor.Ignore(x => x.StringField); 
...

descirptor =>  descriptor.Filter(x => x.StringField);  // Exception
```

**Comment by michaelstaib** [comment]
We also have to do some exception tests on this one... all the argument exceptions are untested.

File: src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs (hunk 2 above)
Comment (PascalSenn) [reply src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs]: done

File: src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs (hunk 1 above)
Comment (PascalSenn) [reply src/Core/Types.Filters/Extensions/FilterFieldCollectionExtensions.cs]: done

**Pull request merged**